### PR TITLE
Adds v1 of program 2023 page registration form

### PR DESCRIPTION
**Pull request by lesleychard**
- Adds new Mailchimp actions to create two student and parent contacts on the new "Participants" list
- Fixes form field labels not showing
- Program information page style and wording clean up

### CODE REVIEW

File: layout/TopNav.tsx
Comment (Freddie-Pike) [inline layout/TopNav.tsx:44]: Appreciate renaming the `hackathon` components since we're no longer using the hackathon branding!

File: lib/mailchimpHelpers.ts
Comment (Freddie-Pike) [inline lib/mailchimpHelpers.ts:5]: Love this utility function!

File: package.json
Comment (Freddie-Pike) [inline package.json:6]: Huh, wonder the package.json shows a diff here? Once my PR is merged, I can pull `main` into this PR for ya to see if this still appears.

File: components/SpeechBubble.tsx
Comment (Freddie-Pike) [inline components/SpeechBubble.tsx:94]: These speech bubbles with the animation delays look flippin' sick btw.

**Review by Freddie-Pike — CHANGES_REQUESTED** [review]
Code reviewed, manually tested. I love the way the program fall 2023 page looks. Label fixes look awesome and again, just love the speech bubbles! I have a few comments from QAing that should be relatively simple to fix below. Sometime during the week I'm going to try to write a playwright test to test out the form here to make it easier to retest this stuff btw and if want me to help ya with anything, lmk! 

1. I think the schema validation for the hook form is a little buggy atm. I get in a state where I can't submit the form despite having no visible errors under the form. Can check the video below.

https://github.com/lesleychard/digital-waves-nl/assets/5024392/15f7dc79-0882-421b-9485-38a3b5decbea

2. Should there a bit more padding between the "Date of Birth" label compared to the other labels? Feels a little tight to me. 

Screenshot 2023-10-08 at 12 08 37 PM

3. The label for the "How did you learn about Digital Waves? *" goes into the select box a bit so I think adding a bit padding there would be nice imo.

Screenshot 2023-10-08 at 12 12 18 PM

File: components/RegistrationForm.tsx
Comment (lesleychard) [reply components/RegistrationForm.tsx:72]: This is copypasta from an old comment - I believe my thinking at the time (2021 so bare with me lol) was that if this form were to be truly "reusable" for any program, the birth year requirements would always be different, so passing it in per program would make sense. Just didn't have time to implement

File: components/RegistrationForm.tsx
Comment (lesleychard) [reply components/RegistrationForm.tsx:224]: LOL yeah this one is quick 'n' dirty

File: components/SpeechBubble.tsx
Comment (lesleychard) [reply components/SpeechBubble.tsx:94]: They're cute hey! Made them a while back, were just hanging out on a local branch for months lol

**Comment by lesleychard** [comment]
Looking at the few other comments above now. The validation bug seems to be the only blocker IMO, but agree with the spacing / label position comments as well. Will fix up if time allows!

File: package.json
Comment (lesleychard) [reply package.json:6]: Ooop - thought I fixed this! I will reset it back to your version in my next commit

**Review by Freddie-Pike — COMMENTED** [review]
There's a build error in Vercel. If it gets updated today, I'll review the PR again tonight. If not, I'll review the PR again tomorrow afternoon 😄

**Review by Freddie-Pike — COMMENTED** [review]
Code reviewed new changes, manually tested. Leaving comment instead of requesting for changes as I'm leaving a few minor testing thoughts I wanted to get your thoughts before approving! Overall, this is soooo awesome Lesley 🎉🎉🎉 

1. When submitting the form, users aren't redirecting to the success call. I added a code suggestion in the code review for a potential fix for this!

2. Even though I have everything filled in correctly, the "Please fix all errors..." message still appears before the form. Wonder if that may be confusing for users who may not realize they can submit the form?

https://github.com/lesleychard/digital-waves-nl/assets/5024392/9f9bf24b-965a-49db-952a-d070e5bb3532

3. I noticed users can submit an empty "other" input field when choosing other for how they found out about digital waves. Should we add schema validation for the other input field to ensure it gets filled in?

https://github.com/lesleychard/digital-waves-nl/assets/5024392/ad88e485-b5e0-47fc-bb47-4729deed1be6

4. This is the most minor thing ever, but just to posting it here for posterity, if a user submits and somehow gets an error page, there's a weird flicker that occurs. Very minor, but just wanted to give ya a heads up there.

https://github.com/lesleychard/digital-waves-nl/assets/5024392/96d6decf-c200-47d8-ba61-73edd2c8af82

File: components/RegistrationForm.tsx
Comment (lesleychard) [reply components/RegistrationForm.tsx:301]: The hash goes to an ID located in `Program2023Register.tsx` component. It likely didn't work for you because you were testing the form multiple times and the hash was already in the browser. I can add a quick fix for this other case next chance I look, not non-blocking issue IMO.

File: components/RegistrationForm.tsx
Comment (lesleychard) [reply components/RegistrationForm.tsx:248]: They _should_ be near the general error message anyways since it's located adjacent to the submit button. Would love if the form scrolled to the correct field with the error, but definitely more effort than we can afford at the moment. Future improvement for sure :D

**Comment by lesleychard** [comment]
Addressing other comments here:

> 2. Even though I have everything filled in correctly, the "Please fix all errors..." message still appears before the form. Wonder if that may be confusing for users who may not realize they can submit the form?

I fixed most of these silly error issues in my latest commit - take a look!

> 3. I noticed users can submit an empty "other" input field when choosing other for how they found out about digital waves. Should we add schema validation for the other input field to ensure it gets filled in?

Non-blocking, but would like to add optional validation at some point for sure.

> 4. This is the most minor thing ever, but just to posting it here for posterity, that if a user submits and somehow gets an error page, there's a weird flicker that occurs. Very minor, but just wanted to give ya a heads up there haha.

Think it's because the section w/ the white background extends when the error message is shown. Easy fix would be to add enough padding under the submit button to render the error absolute positioned. Future consideration 😅

**Comment by lesleychard** [comment]
Also generally speaking `RegistrationForm` has gotten out of control. A great future improvement would be to split the 3 sections of the form into separate sub-components and maybe even implement a stepper so it doesn't look so intimidating to fill out! We'll get there someday 😄

**Comment by Freddie-Pike** [comment]
> Also generally speaking `RegistrationForm` has gotten out of control. A great future improvement would be to split the 3 sections of the form into separate sub-components and maybe even implement a stepper so it doesn't look so intimidating to fill out! We'll get there someday 😄

I would love to do that next year if you wouldn't mind reviewing my PR there? I want to get more experience with Figma so potentially making a design for a multi stepper for you to review would be super cool!

**Review by Freddie-Pike — COMMENTED** [review]
Code reviewed, manually tested. PR overall looks awesome! Only minor comment is there isn't phone number format validation on the phone number. We could add something like `google-libphonenumber` as a stretch goal and use that with Yup or use as a regex. That's a stretch goal though as phone validation is just really dicey from my experience with it and it's a minor comment. 

PR looks great, left a comment RE: grammar I wasn't sure about. Once this PR is ready, I'll review again and if it looks good again, I'll approve 😄

File: modules/program-2023/Program2023FAQ.tsx
Comment (lesleychard) [reply modules/program-2023/Program2023FAQ.tsx:15]: Good catch - the dashes themselves are grammatically correct, but there should not be a space after `trans-`. I will fix now.

**Comment by lesleychard** [comment]
> Code reviewed, manually tested. PR overall looks awesome! Only minor comment is there isn't phone number format validation on the phone number. We could add something like `google-libphonenumber` as a stretch goal and use that with Yup or use as a regex. That's a stretch goal though as phone validation is just really dicey from my experience with it and it's a minor comment.
> 
> PR looks great, left a comment RE: grammar I wasn't sure about. Once this PR is ready, I'll review again and if it looks good again, I'll approve 😄

Validation would be a nice-to-have for sure in future PR. Mailchimp parses whatever format the users input to a consistent format on their end through my testing which is nice in the interim :)

File: modules/program-2023/Program2023Hero.tsx
Comment (lesleychard) [reply modules/program-2023/Program2023Hero.tsx:41]: It's grammatically correct :) just not so commonly used because the situation doesn't often present itself haha

**Comment by lesleychard** [comment]
Actually @Freddie-Pike the FAQ section w/ the grammar error isn't in use right now so I'm going to leave it. Please give me ✅ if all good with you so I can get this live tonight!

**Comment by lesleychard** [comment]
@Freddie-Pike I'm going live now with an implied ✅ before I run out of steam tonight. Appreciate all the help with this page!!! ❤️

**Pull request merged**
